pkg/trivyk8s/trivyk8s.go

```diff
@@ -0,0 +1,99 @@
+package trivyk8s
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/aquasecurity/trivy-kubernetes/pkg/artifacts"
+	"github.com/aquasecurity/trivy-kubernetes/pkg/k8s"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// GetKubeConfig returns local kubernetes configurations
+func GetKubeConfig() (*rest.Config, error) {
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+
+	kubeConfigPath := filepath.Join(userHomeDir, ".kube", "config")
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return kubeConfig, nil
+}
+
+// TrivyK8S interface represents the operations supported by the library
+type TrivyK8S interface {
+	ListArtifacts(context.Context, string) ([]*artifacts.Artifact, error)
+}
+
+type client struct {
+	k8s dynamic.Interface
+}
+
+// New creates a trivyK8S client
+func New(kubeConfig *rest.Config) (TrivyK8S, error) {
```

Comment:
I would inject the dynamic client on to this constructor; it could help with testing (using fake client).

Reply:
Yes, I'm using the fake client on the tests.

Reply:
I assume that you'll have to change the New() function signature to accept client as well